packages/neon/neon/lib/src/settings/models/option.dart

```diff
@@ -1,63 +1,114 @@
 import 'dart:async';
 
+import 'package:flutter/foundation.dart';
 import 'package:meta/meta.dart';
 import 'package:neon/src/settings/models/options_category.dart';
 import 'package:neon/src/settings/models/storage.dart';
 import 'package:neon/src/settings/widgets/label_builder.dart';
 import 'package:rxdart/rxdart.dart';
 
 @internal
-class OptionDisableException implements Exception {}
-
-@immutable
-@internal
-abstract class Option<T> {
+abstract class Option<T> extends ChangeNotifier implements ValueListenable<T> {
+  /// Creates an Option
   Option({
     required this.storage,
     required this.key,
     required this.label,
     required this.defaultValue,
-    required this.stream,
+    final bool enabled = true,
     this.category,
-    final BehaviorSubject<bool>? enabled,
-  }) : enabled = enabled ?? BehaviorSubject<bool>.seeded(true);
+    final T? initialValue,
+  })  : _value = initialValue ?? defaultValue,
+        _enabled = enabled;
+
+  /// Creates an Option depending on the State of another one.
+  Option.depend({
```

Comment:
I know I suggested this name, but somehow I still don't like it :see_no_evil: I don't have something better though

Reply:
Maybe rely?

But I not too convinded.

Reply:
maybe dependent? we can just leave it and fix it later if needed